src/controller/ContactController.php

```diff
@@ -107,14 +96,20 @@ public function displayContactPage(): void
      */
     public function manageContact(string $name, string $firstName, string $email, string $content): void
     {
-        if (!$this->isSubmitted(self::ACTION) || !$this->isValid($_POST)) {
+        $post = [
+            "name" => $name,
+            "firstName" => $firstName,
+            "email" => $email,
+            "content" => $content
+        ];
+        if (!$this->isValid($post)) {
             $template = RouteMapper::ContactView->getTemplate();
             $data = [
                 MessageMapper::Error->getMessageLabel() => MessageMapper::GeneralError->getMessage()
             ];
         }
 
-        if (!isset($data[MessageMapper::Error->getMessageLabel()])) {
+        if (isset($data[MessageMapper::Error->getMessageLabel()]) === false) {
             $currentDate = DateTime::createFromFormat("Y-m-d H:i:s", date("Y-m-d H:i:s"));
             $contact = new ContactModel($name, $firstName, $email, $content, $currentDate);
             $validateContact = $this->validContactForm($contact);
```

Comment:
je pense qu'ici tu travailles sur la même instance de $contact. Du coup tu pourrais simplement ne pas faire de return à ta fonction validContactForm.
```suggestion
            $this->validContactForm($contact);
            $contactCreated = $this->_contactService->createContact($contact);
```

Reply:
et ensuite dans ton `if($contactCreated) `

faire : 
```suggestion
            $contact->setContent(htmlspecialchars_decode($contact->getContent()));
            $sendMail = $this->_contactService->notify($contact);
```

Mais du coup à quoi sert ton $contactCreated ? c'est un boolean ? si oui je le renommerais en `$isContactCreated`

Reply:
et je rajouterais juste avant le `if($contactCreated)` 
```suggestion
            $sendMail = false
```

Reply:
ok fait